go/vt/wrangler/vexec_plan.go

```diff
@@ -17,48 +17,186 @@ limitations under the License.
 package wrangler
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 	"vitess.io/vitess/go/vt/sqlparser"
+
+	"github.com/olekukonko/tablewriter"
 )
 
+// vexecPlan contains the final query to be sent to the tablets
 type vexecPlan struct {
-	query       string
 	opcode      int
 	parsedQuery *sqlparser.ParsedQuery
 }
 
+// vexecPlannerParams controls how some queries/columns are handled
+type vexecPlannerParams struct {
+	dbNameColumn         string
+	workflowColumn       string
+	immutableColumnNames []string
+	updatableColumnNames []string
+	updateTemplates      []string
+}
+
+// vexecPlanner generates and executes a plan
+type vexecPlanner interface {
+	params() *vexecPlannerParams
+	exec(ctx context.Context, masterAlias *topodatapb.TabletAlias, query string) (*querypb.QueryResult, error)
+	dryRun() error
+}
+
+// vreplicationPlanner is a vexecPlanner implementation, specific to _vt.vreplication table
+type vreplicationPlanner struct {
+	vx *vexec
+	d  *vexecPlannerParams
+}
+
+func newVreplicationPlanner(vx *vexec) vexecPlanner {
```

Comment:
newVreplicationPlanner should be newVReplicationPlanner. General rule we have been following is to also capitalize R when V is capitalized as part of VReplication ...

Reply:
fixed!